### PR TITLE
Sum even and odd

### PR DESCRIPTION
**Pull request by ermishina**
---
name: solution review 
about: A template PR for code review with a checklist
---

### Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
  - [x] There is no logic in the unit test
- [x] All tests pass
- [x] There are tests for defensive assertions
- [ ] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [x] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [x] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [x] Each assumption about an argument is checked with an assertion
  - [x] Each assertion checks for _only one_ assumption about the argument
- [ ] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [ ] The function has correct type annotations
- [x] The function is not called in the function file

### Do's

- [x] Variable names help to understand the strategy
- [ ] Any comments are clear and describe the strategy
- [x] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [x] The function's strategy _is not_ described in the documentation
- [ ] Comments explain the _strategy_, **not** the _implementation_
- [x] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [x] The code passes the formatting checks
- [x] The code passes all Ruff linting checks
- [x] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [x] Variables are named with snake_case
- [x] Variable names are clear and helpful
- [ ] The code follows the strategy as simply as possible
- [x] The implementation is as simple as possible given the strategy
- [x] There are no commented lines of code
- [x] There are no `print` statements anywhere
- [ ] The code includes defensive assertions
- [ ] Defensive assertions include as little logic as possible

### CODE REVIEW

File: .github/workflows/ci-checks.yml [hunk 1]
```diff
@@ -76,5 +76,5 @@ jobs:
       
       - name: Python - Run Tests
         if: steps.check_tests.outputs.has_tests == 'true'
-        run: python -m unittest
```
Comment (Azizsin7) [inline .github/workflows/ci-checks.yml:79]: the test file is working without problem, can you explain why did you add this ci check?

File: solutions/README.md [hunk 2]
```diff
@@ -1 +1 @@
-# Solutions
+# Tests
```
Comment (Azizsin7) [inline solutions/README.md:1]: this readme is for the solution, there is another readme in test folder to add the tests

File: solutions/sum_even_and_odd.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
+"""This module provides a function sum_even_and_odd
+that calculates the sums of positive and negative,
+even and odd numbers in a given list.
+"""
+
+
+def sum_even_and_odd(numbers):
```
Comment (Azizsin7) [inline solutions/sum_even_and_odd.py:7]: add annotations

File: solutions/sum_even_and_odd.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+    Raises:
+        ValueError: If input is not a list or contains non-numeric values.
+
+    Examples:
+    >>> sum_even_and_odd([1, -2, 3, 4, -5, 0])
+    {'positive_even': 4, 'positive_odd': 4, 'negative_even': -2, 'negative_odd': -5}
+    >>> sum_even_and_odd([])
+    {'positive_even': 0, 'positive_odd': 0, 'negative_even': 0, 'negative_odd': 0}
+    """
```
Comment (Azizsin7) [inline solutions/sum_even_and_odd.py:26]: Include 3 or more (passing!) doctests
Edge Cases

File: solutions/sum_even_and_odd.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+            raise ValueError("All elements must be integers or floats.")
+
+    result = {
+        "positive_even": 0,
+        "positive_odd": 0,
+        "negative_even": 0,
+        "negative_odd": 0,
+    }
+
+    for num in numbers:
```
Comment (Azizsin7) [inline solutions/sum_even_and_odd.py:41]: Floating-point number into the function is not properly handled

File: solutions/sum_even_and_odd.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
+"""This module provides a function sum_even_and_odd
```
Comment (Azizsin7) [inline solutions/sum_even_and_odd.py:1]: Add:
Shebang Line: #!/usr/bin/env python3
Encoding Declaration: # -*- coding: utf-8 -*-

File: solutions/tests/test_sum_even_and_odd.py [hunk 7]
```diff
@@ -0,0 +1,83 @@
+"""
```
Comment (Azizsin7) [inline solutions/tests/test_sum_even_and_odd.py:1]: Add:
Shebang Line: #!/usr/bin/env python3
Encoding Declaration: # -- coding: utf-8 --

File: solutions/tests/test_sum_even_and_odd.py [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+This module contains unit tests for the `sum_even_and_odd` function.
+
+The tests ensure that the function correctly calculates the sums of
+positive even, positive odd, negative even, and negative odd numbers in a list.
+"""
+
+import unittest
+
+from solutions.sum_even_and_odd import sum_even_and_odd  # Import the function to test
```
Comment (Azizsin7) [inline solutions/tests/test_sum_even_and_odd.py:10]: import like:

from ..sum_even_and_odd import sum_even_and_odd

**Review by Azizsin7 — CHANGES_REQUESTED** [review]
**Your sum_even_and_odd.py needs:**

> Shebang Line: #!/usr/bin/env python3
> Encoding Declaration: # -- coding: utf-8 --
> 
> add annotations
> 
> Include 3 or more (passing!) doctests
> Edge Cases
> 
> Floating-point number into the function is not properly handled

**Your test file needs:**

> Shebang Line: #!/usr/bin/env python3
> Encoding Declaration: # -- coding: utf-8 --
> 
> import method
> 
> Boundary tests
> 
> More defensive tests like mixed types
